### DISCUSSION TITLE
stm32 fast adc sampling rate

### DISCUSSION THREAD

**Discussion by codev123** [stm32duino/Arduino_Core_STM32#2234] [Q&A] [answered]
sir, thank you for amazing platform. i am reading analog signal using 
unsigned long time_start = micros();
	for (int i = 0; i < 100; i++){
		temp = analogRead(PA0);
	}
	unsigned long time_end = micros();
but for 100 samples its taking 6616uS while as advertized it should take 1-6uS per sample not 66uS per sample. how to make this sampling faster for stm32l051c8 and stm32f103c8 stm microcontrollers.
thank you.

**Comment by JingtaiChen** [comment]
I had the same question, I asked the official and the official answer was to use the HAL

**Comment by fpistm** [comment, marked as answer]
> I had the same question, I asked the official and the official answer was to use the HAL

Currently, this is the only way.
Here an example how to convert a STM32Cube project to get faster ADC read:
https://www.stm32duino.com/viewtopic.php?t=110
